File: 5-classes.rb
# Ruby 用 attr_reader 暴露只讀屬性，instance variable 會用 @ 開頭。
class User
  attr_reader :name, :role

  def initialize(name, role)
    @name = name
    @role = role
  end

  def admin?
    role == 'admin'
  end
end

user = User.new('Ada', 'admin')

puts user.name
puts user.role
puts user.admin?

File: 5-classes.py
# Python class 會經 self 擺屬性同呼叫 method，寫法比較顯式。
class User:
    def __init__(self, name, role):
        self.name = name
        self.role = role

    def is_admin(self):
        return self.role == "admin"


user = User("Ada", "admin")

print(user.name)
print(user.role)
print(user.is_admin())
